### PR TITLE
[ios][android][expo] Polyfill crypto.getRandomValues()

### LINKED ISSUE #7209: Compatibility with uuid 7.x, nanoid, and others
### Environment

```
  Expo CLI 3.13.1 environment info:
    System:
      OS: macOS 10.15.2
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.12.0 - /usr/local/bin/node
      Yarn: 1.21.1 - /usr/local/bin/yarn
      npm: 6.13.6 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 3.13.1
```

### Steps to Reproduce

There are a lot of libraries that requires synchronous random values. Many of those libraries are moving away from `Math.random()` because of the numerous issues with them.

The new standard web platforms api is the `crypto.getRandomValues` API, and this is what e.g. uuid and nanoid uses. This is an excellent API that gives you secure random values. The newly released version 7.x of uuid does no longer work without this API.

I've previously tried to [add `crypto.getRandomValues` into React Native itself](https://github.com/facebook/react-native/pull/20686), but unfortunately that never came to fruition. I still think that that would be the absolut best approach.

The next best approach for vanilla React Native is a polyfill! So that's why I wrote: [react-native-get-random-values](https://github.com/LinusU/react-native-get-random-values).

Unfortunately this polyfill doesn't work in Expo since it requires native code.

My proposal is to add the package to Expo as with other native packages, but also go one step further and adding the `require` of the polyfill to Expo itself, so that `crypto.getRandomValues()` will just work out of the box for everyone!

-----

We've already heard from a few users that are having problem with uuid in React Native/Expo, see some previous discussions: https://github.com/uuidjs/uuid/issues/375 https://github.com/uuidjs/uuid/pull/385

What we did for now was add instructions to React Native users to use the polyfill, but unfortunately we cannot offer a solution to Expo users right now 😱 

-----

As always I'm willing to submit PRs for this 👍

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

Fixes https://github.com/expo/expo/issues/7209.

# How

Include [react-native-get-random-values](https://github.com/LinusU/react-native-get-random-values) and import it in Expo.fx.expo.ts and Expo.fx.ts, so it will be polyfilled by default in managed and bare workflow apps created with expo-cli or create-react-native-app.

One intentional and possibly controversial choice here is to polyfill the library automatically. I did this because the cost to users who don't use it is rounds to zero, and when users do want to use it, it's a much better experience to just have libraries like `uuid` work out of the box. If we decide to switch the implementation details on how the polyfill works, we can do that easily at some point because the users do not explicitly import anything or call any methods to polyfill.

I experimented with switching expo-random methods to synchronous methods, which required converting from a unimodule to a plain React Native module. This was pretty easy on iOS if I left unimodule.json in place, but on Android it led to some special casing for Expo packages that live in the /packages directory and are not unimodules and I didn't like how that ended up.

# Test Plan

Try out the "Polyfills" screen in NCL.

### CODE REVIEW

**Comment by brentvatne** [comment]
CI shows iOS build fails, but the logs end with "** BUILD SUCCEEDED **" 🤔

File: apps/native-component-list/package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "expo-print": "~9.0.2",
     "expo-processing": "2.0.2",
     "expo-progress": "^0.0.2",
+    "expo-random": "^8.2.1",
```
Comment (EvanBacon) [inline apps/native-component-list/package.json:66]: why add

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/getrandomvalues/RNGetRandomValuesModule.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Override
+  public String getName() {
+    return "RNGetRandomValues";
+  }
+
+  @ReactMethod(isBlockingSynchronousMethod = true)
+  public String getRandomBase64(int byteLength) throws NoSuchAlgorithmException {
+    byte[] data = new byte[byteLength];
+    SecureRandom random = new SecureRandom();
```
Comment (EvanBacon) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/getrandomvalues/RNGetRandomValuesModule.java:30]: This native code already exists in the SDK https://github.com/expo/expo/blob/master/packages/expo-random/android/src/main/java/expo/modules/random/RandomModule.java just in an async manner. Adding this duplicative code essentially just ends up being code bloat as one of the methods would be unused.

Comment (brentvatne) [reply android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/getrandomvalues/RNGetRandomValuesModule.java:30]: reasoning in description of pr

**Comment by brentvatne** [comment]
need to rethink this approach, won’t work well when ejecting to bare actually

**Pull request closed without merging**